CMakeLists.txt

```diff
@@ -244,60 +244,60 @@ endif()
 include(cmake/optional.cmake)
 
 # Don't allow extended (non-ASCII) symbols in identifiers. This is easier for code review.
-try_append_cxx_flags("-fno-extended-identifiers" TARGET core)
+try_append_cxx_flags("-fno-extended-identifiers" TARGET core_interface)
 
 # Currently all versions of gcc are subject to a class of bugs, see the
 # gccbug_90348 test case (only reproduces on GCC 11 and earlier) and
 # https://gcc.gnu.org/bugzilla/show_bug.cgi?id=111843. To work around that, set
 # -fstack-reuse=none for all gcc builds. (Only gcc understands this flag).
-try_append_cxx_flags("-fstack-reuse=none" TARGET core)
+try_append_cxx_flags("-fstack-reuse=none" TARGET core_interface)
 
 if(HARDENING)
-  add_library(hardening INTERFACE)
+  add_library(hardening_interface INTERFACE)
+  target_link_libraries(core_interface INTERFACE hardening_interface)
```

Comment:
Why is this moved up here?

Reply:
To make both lines visually connected. The benefit of interface libraries is that the `target_link_libraries` command might be placed anywhere after the `add_library` one.